### PR TITLE
Added mechanism to make IN124 fail only on newly hidden params

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes: [CIAC-10734](https://jira-dc.paloaltonetworks.com/browse/CIAC-10734)

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/IN_validators/IN124_is_hiddenable_param.py [hunk 1]
```diff
@@ -83,13 +77,37 @@ def get_invalid_hidden_params(
...
+            content_item (Integration): The integration to test.
+            param (Parameter): The current param to check.
+
+        Returns:
+            bool: True if the param was already set to True in the old content object. Otherwise, return False.
+        """
+        return bool(
+            (old_obj := content_item.old_base_content_object)
+            and content_item.old_base_content_object.git_status
+            in [GitStatuses.MODIFIED, GitStatuses.RENAMED]
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/IN_validators/IN124_is_hiddenable_param.py:105]: why arent these added to the head of the file like GItStatuses = []..?

File: demisto_sdk/commands/validate/tests/IN_validators_test.py [hunk 2]
```diff
@@ -4376,6 +4382,42 @@ def test_IsHiddenableParamValidator_is_valid(
...
+        - Case 1: Should pass all except for the third content item.
+    """
+    content_items = INVALID_HIDDEN_PARAM_INTEGRATIONS
+    old_content_items = copy.deepcopy(content_items)
+    for old_content_item, git_status in zip(
+        old_content_items,
+        [
+            GitStatuses.MODIFIED,
+            GitStatuses.RENAMED,
+            GitStatuses.ADDED,
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/tests/IN_validators_test.py:4407]: why should it run on added?

File: demisto_sdk/commands/validate/tests/IN_validators_test.py [hunk 3]
```diff
@@ -4376,6 +4382,42 @@ def test_IsHiddenableParamValidator_is_valid(
...
+            GitStatuses.MODIFIED,
+        ],
+    ):
+        old_content_item.git_status = git_status
+    create_old_file_pointers(content_items, old_content_items)
+    results = IsHiddenableParamValidator().is_valid(content_items)
+    assert len(results) == 1
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/tests/IN_validators_test.py:4414]: lets add test cases for the flow it shouldnt fail because it was already hidden

**Review by JudahSchwartz — APPROVED** [review]
nice! note my comments

File: demisto_sdk/commands/validate/validators/IN_validators/IN124_is_hiddenable_param.py (hunk 1 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validators/IN_validators/IN124_is_hiddenable_param.py:105]: Good point tbh...

File: demisto_sdk/commands/validate/tests/IN_validators_test.py (hunk 3 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/tests/IN_validators_test.py:4414]: That's exactly what the new test does.

**Pull request merged**